controllers/namespace/default_deny_network_policy.go

```diff
@@ -88,13 +82,26 @@ func (r *NamespaceReconciler) reconcileDefaultDenyNetworkPolicy(ctx context.Cont
 						},
 					},
 				},
-				// Egress rule for instana-agents
+				// Egress rule for grafana-agent
 				{
 					To: []networkingv1.NetworkPolicyPeer{
 						{
-							IPBlock: &networkingv1.IPBlock{
-								CIDR: instanaConfigMap.Data["cidrBlock"],
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"kubernetes.io/metadata.name": "grafana-agent"},
 							},
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"app": "grafana-agent"},
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: util.PointTo(corev1.ProtocolTCP),
+							Port:     util.PointTo(intstr.FromInt(4317)),
```

Comment:
where do the port numbers come from? haven't had a lot to do with grafana-agent

Reply:
From here https://kartverket.atlassian.net/browse/SKIP-1164